### PR TITLE
Rewrite Keyboard.cpp in C

### PR DESCRIPTION
**Pull request by riidefi**

### CODE REVIEW

File: payload/sp/IOSKeyboard.h [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+    IOS_KEY_TAB = 43,
+    IOS_KEY_SPACE = 44,
+} IOSKeyboard_KeyCode;
+
+#if 0
+inline IOSKeyboard_KeyCode IOSKeyboard_LetterToKeycode(char c) {
+    assert((c >= 'a' && c <= 'z') || (c >= 'A' && c <= 'Z'));
+    return IOSKeyboard_KeyCode(c >= 'a' ? (c - 'a' + IOS_KEY_A) : c - 'A' + IOS_KEY_A);
+}
+#endif
```
Comment (stblr) [inline payload/sp/IOSKeyboard.h:37]: Do we need this?

File: payload/sp/IOSKeyboard.c [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+    '?',   // unknown
+    '\\',  // right \ PAD
+    ';', '\'', '~', ',', '.', '/',
+    0  // CAPS = 57
+};
+static const char keys_shift[] = {
+    0, 0, 0, 0, 'A', 'B', 'C', 'D', 'E', 'F', 'G', 'H', 'I', 'J', 'K', 'L', 'M', 'N', 'O',
+    'P', 'Q', 'R', 'S', 'T', 'U', 'V', 'W', 'X', 'Y',
+    'Z',  // 29
+    '!',  // 31 -- This is wrong on wiibrwe, 30
```
Comment (stblr) [inline payload/sp/IOSKeyboard.c:34]: brew

File: configure.py [hunk 3]
```diff
@@ -223,7 +223,8 @@
         os.path.join('nw4r', 'lyt', 'lyt_layout.S'),
         os.path.join('nw4r', 'snd', 'snd_DvdSoundArchive.S'),
         os.path.join('revolution', 'nand.c'),
-        os.path.join('sp', 'Keyboard.cpp'),
+        os.path.join('sp', 'IOSKeyboard.c'),
+        os.path.join('sp', 'Keyboard.c'),
         os.path.join('sp', 'Fatal.c'),
```
Comment (stblr) [inline configure.py:228]: These should be sorted alphabetically.

**Review by stblr — CHANGES_REQUESTED** [review]

**Comment by riidefi** [comment]
Fixed

**Review by stblr — APPROVED** [review]

**Pull request merged**